src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenExprLambdaTests.cs

```diff
@@ -381,6 +381,42 @@ public static string GetCultureInvariantString(object value)
 ";
         #endregion A string containing expression-tree dumping utilities
 
+        [Fact]
+        public void ExprLambdaIndexerCompoundAssignment()
+        {
+            var verifier = CompileAndVerify(@"
+using System;
+using System.Linq.Expressions;
+
+class C
+{
+    private static Expression<Func<int>> _f1;
+    private static Expression<Func<int>> _f2;
+
+    public static void Main() {
+        var c = new C();
+        c[() => 0] += 1;
+        Console.WriteLine(object.ReferenceEquals(_f1, _f2));
+    }
+
+    int this[Expression<Func<int>> f] {
```

Comment:
>Expression<Func<int>> [](start = 13, length = 21)

If this test is in response to my comment about expression trees, this isn't the scenario. The scenario is when the expression that uses lambda as an argument is converted to an expression tree, not just the lambda itself

Reply:
We now put lambda or a delegate in a temp in some cases, that might affect our ability to create an expression tree.

---
In reply to: [179239269](https://github.com/dotnet/roslyn/pull/25917#discussion_r179239269) [](ancestors = 179239269)

Reply:
Ah! Thanks I will try to write such a test.

Reply:
Note that you change affects more scenarios, not just compound assignments. 

---
In reply to: [179240865](https://github.com/dotnet/roslyn/pull/25917#discussion_r179240865) [](ancestors = 179240865)